editor/doc/http-api.md

```diff
@@ -0,0 +1,98 @@
+Editor HTTP API
+===============
+
+External tools can connect to the running editor and interact with it using a REST API.
+
+### Caveats
+* This HTTP API is currently in an experimental state. It might change drastically or even be removed completely if we deem it necessary. But hopefully it shouldn't come to that.
+* Currently, the port is assigned randomly from a broad subrange. The editor will log the local endpoint url if started from a terminal.
```

Comment:
> The editor will log the local endpoint url if started from a terminal.

If VS Code started the editor from a terminal could it read the port from logs somehow?

Reply:
Yes, that should be possible. I'm also thinking that maybe VSCode could set an env variable with a port for the editor to use. If the env variable is set the editor will use it, otherwise it will use a random port. Similar to the DM_SERVICE_PORT between editor and engine:

https://github.com/defold/defold/blob/dev/engine/docs/DEBUG_PORTS_AND_SERVICES.md

Reply:
Which document?

Reply:
The link britzl posted?

Reply:
Although I'm a bit annoyed that the example checks for any service that contains the substring `"Defold"`. Where does that leave our "Defold Editor" service? 🤔

Reply:
It's like the Windows 8 -> Windows 10 situation where they skipped windows 9 because of `iswin95or98=os.startsWith("Windows 9")` :D